src/lib/registration/sw_mgmt.rb

```diff
@@ -653,8 +654,8 @@ def self.select_product_addons(products, addons)
     # find the product resolvables from the specified repository
     def self.products_from_repo(repo_id)
       # TODO: only installed products??
-      Pkg.ResolvableProperties("", :product, "").select do |product|
-        product["source"] == repo_id
+      Y2Packager::Resolvable.find(kind: :product).select do |product|
```

Comment:
You can add more filters to the find call:

```suggestion
      Y2Packager::Resolvable.find(kind: :product, source: repo_id)
```